src/athena.js

```diff
@@ -157,15 +162,26 @@ export let executeQuery = async function({
     throw Error("Athena: query didn't succeed.");
   }
 
-  let queryResult = await athena.getQueryResults({QueryExecutionId}).promise();
+  let queryResult;
+  let nextToken;
+  let rows = [];
+  do {
+    queryResult = await athena.getQueryResults({QueryExecutionId, NextToken: nextToken}).promise();
 
-  if (queryResultIsShowResult(queryResult)) {
-    return queryResultToText(queryResult);
-  }
+    // checking if the query is a result of SHOW query
+    // returing just text in this case, not trying to parse columns and rows
+    if (queryResultIsShowResult(queryResult)) {
+      return queryResultToText(queryResult);
+    }
+
+    nextToken = queryResult.NextToken;
+
+    rows = rows.concat(queryResultToObjectsArray(queryResult));
+  } while (nextToken && rows.length <= maxRows);
```

Comment:
don't you mean `maxRows + 1` (column names) ?

Reply:
or actually to `_.drop(rows, 1)` immediately after getting the first page (if you split the code like I wrote above)

Reply:
> maxRows + 1

that's why there is `rows.length <= maxRows` instead of `rows.length < maxRows + 1`. But I agree that it could be cleaner.